Wrappers/Python/cil/processors/MaskGenerator.py

```diff
@@ -0,0 +1,334 @@
+            
+            mask[numpy.logical_or(data.as_array() < threshold[0], data.as_array() > threshold[1])] = 0
+            
+        elif self.mode == 'quantile':
+            
+            if not(isinstance(self.quantiles, tuple)):
+                raise Exception("Quantiles must be given as a tuple containing two values,\n " + \
+                    "use None if no quantile value is given")
+            
+            quantile = self._parse_quantile_value(data)
+                
+            mask[numpy.logical_or(data.as_array() < quantile[0], data.as_array() > quantile[1])] = 0
+        
+        elif self.mode == 'mean':
+            
+            # if mean along specific axis
+            if self.axis is not None:
+                
+                axis = data.get_dimension_axis(self.axis)
+                ndim = len(data.dimension_labels)
+                
+                tile_par = []
+                slice_obj = []
+                for i in range(ndim):
+                    if i == axis:
+                        tile_par.append(data.get_dimension_size(self.axis))
+                        slice_obj.append(numpy.newaxis)
+                    else:
+                        tile_par.append(1)
+                        slice_obj.append(slice(None, None, 1))
+                tile_par = tuple(tile_par)
+                slice_obj = tuple(slice_obj)
+                
+                tmp_mean = numpy.tile((numpy.mean(data.as_array(), axis=axis))[slice_obj], tile_par)
+                tmp_std = numpy.tile((numpy.std(data.as_array(), axis=axis))[slice_obj], tile_par)
+                mask[numpy.abs(data.as_array() - tmp_mean) > self.threshold_factor * tmp_std] = 0
+
+            # if global mean    
+            else:
+                
+                 mask[numpy.abs(data.as_array() - numpy.mean(data.as_array())) > 
+                      self.threshold_factor * numpy.std(data.as_array())] = 0
+        
+        elif self.mode == 'median':
+            
+            c = -1 / (numpy.sqrt(2) * special.erfcinv(3 / 2))
+            
+            # if median along specific axis
+            if self.axis is not None:
+
+                axis = data.get_dimension_axis(self.axis)
+                ndim = len(data.dimension_labels)
+                
+                tile_par = []
+                slice_obj = []
+                for i in range(ndim):
+                    if i == axis:
+                        tile_par.append(data.get_dimension_size(self.axis))
+                        slice_obj.append(numpy.newaxis)
+                    else:
+                        tile_par.append(1)
+                        slice_obj.append(slice(None, None, 1))
+                tile_par = tuple(tile_par)
+                slice_obj = tuple(slice_obj)
+
+                tmp = numpy.abs(data.as_array() - numpy.tile((numpy.median(data.as_array(), axis=axis))[slice_obj], tile_par))
+                MAD = numpy.tile((numpy.median(tmp, axis=axis))[slice_obj], tile_par)
```

Comment:
I'm not sure what `MAD` is or why it's in syntax highlighting, but maybe it's a reserved python keyword? I'd use a more descriptive variable name for maintainability.